### DISCUSSION TITLE
Ring & HASS stopped working - not publishing the entities

### DISCUSSION THREAD

**Discussion by talshum** [tsightler/ring-mqtt#607] [Support Q&A] [answered]
Hi! thank you for all the effort with this add on.
It has been working for me great for years. Recently after an update of HASS the integration stopped working for me.
I see all the logs as if it recognizes my ring alarm entities but I don't see it in the devices/entity section.
I do see logs like the following which made me think it should work .

Thank you

2023-04-06T16:39:55.007Z ring-mqtt [Contact Front Door] ring/10c655c2-a35e-4127-bcb0-cedf1b1f1a8e/alarm/e75dd2a6-af8f-442e-801d-db41ff370bfd/contact/state OFF
2023-04-06T16:39:55.013Z ring-mqtt [Contact Front Door] ring/10c655c2-a35e-4127-bcb0-cedf1b1f1a8e/alarm/e75dd2a6-af8f-442e-801d-db41ff370bfd/bypass_mode/state Never
2023-04-06T16:39:55.017Z ring-attr [Contact Front Door] ring/10c655c2-a35e-4127-bcb0-cedf1b1f1a8e/alarm/e75dd2a6-af8f-442e-801d-db41ff370bfd/info/state {"batteryLevel":100,"batteryStatus":"full","commStatus":"ok","firmwareStatus":"up-to-date","lastCommTime":"2023-04-06T16:39:53Z","lastUpdate":"2023-04-06T16:39:53Z","linkQuality":"ok","serialNumber":"BHDW11852EY237279","tamperStatus":"ok"}

**Comment by tsightler** [comment]
Please verify that you have rebooted the system.  Not just restarted, but, once you click restart in HA, select advanced options and choose reboot host.  Unfortunately, if ring-mqtt itself is working, as the logs indicate, then the problem has to be on the HA side, and I can't be much help there, but a proper a full reboot, not the a restart, fixes 99% of reports like this.

**Comment by talshum** [comment]
thank you for the quick reply. Unfortunately even taking the power out didn't help.
Your logs show that if there is any offline device, nothing will be published. I removed the offline device but still no luck. I can dig into the code - any clue which part of hass is taking care of this?

**Reply by tsightler** [reply]
The logs you posted above indicate that ring-mqtt itself is connected to the Ring API and to the MQTT broker and is sending messages.  In that state, the only thing that can be wrong is the HA side which receives messages from MQTT.  You can of course connect to the MQTT broker with any MQTT client and monitor the messages.   Overall it's not complex, ring-mqtt connect to Ring API, send message to MQTT broker, HA connect to MQTT broker via Mosquitto integration and listens for messages.  But ring-mqtt can only know if it is sending messages, it cannot know if HA or any other MQTT client is actually receiving them.

**Reply by talshum** [reply]
thanks, I was able to 'receive' the updates from mqttt on Node-Red. But I'm not sure how to log through HA? How do I make sure HA knows how to read the pub context and make entities out of it?

**Reply by tsightler** [reply]
Those are HA questions.  I apologize, but I only support my project, I don't really know that much about the internal workings of HA, other than the fact that it is the core-mosquitto integration which provides HA with the connection to the MQTT broker. You might just try uninstalling, restarting and letting that integration be rediscovered and then reconfigure that component.  Otherwise, your best bet is to ask on HA forums.

**Reply by talshum** [reply, marked as answer]
Update here:
Looks like the HA update broke some stuff.
I deleted and re-install Mosquitto broker and also removed the old config from the configurator.yaml
then reinstall the ha broker through the add on.
magic...

thanks so much!

**Reply by tsightler** [reply]
Thanks for sharing your result and resolution!
